packages/remote-config/e2e/config.modular.e2e.js

```diff
@@ -743,5 +749,14 @@ describe('remoteConfig() modular', function () {
         should(supported).equal(true);
       });
     });
+
+    describe('onConfigUpdated', function () {
+      it('onConfigUpdated can run without an issue', async function () {
+        const { getRemoteConfig, onConfigUpdated } = remoteConfigModular;
+        const remoteConfig = getRemoteConfig();
+
+        onConfigUpdated(remoteConfig, function (event, error) {});
```

Comment:
I think this test need to be completed

Reply:
I couldn't think of a test here. Followed tests/integration_test/firebase_remote_config/firebase_remote_config_e2e_test.dart in this case. In the comment there, you mentioned that it's not possible to change values on the fly to test the callback. Any ideas on how to improve this test?

Reply:
I think the tests are inadequate and I'm working on a test style that uses a cloud function I will deploy (I'm working on this currently, nearly ready).

The cloud function will have access to the admin SDK which allows it to read/write the remote-config template, and will have an API that lets you add a remote config template key, remove a remote config template key, and update a remote config template key value as a list of 0 or more operations that it will do at once.

So then in the e2e tests I expect this to happen:

- single listener addition/removal case:
  - we test updating a known remote-config template key value while there are no listeners, no listeners are called
  - we add a listener, do same update, listener should be called
  - we remove the listener, do same update, no listener should be called

- update type verification
  - make sure key addition is handled correctly
  - same with key removal
  - same with key update of all update types

- react-native reload verification
  - add a listener, verify it is called when you update the remote-config template
  - do a react-native hot reload, make sure the listener is removed down to native level

- multiple listener addition/removal accounting verification
  - add one listener, make sure it is called when template updates
  - add second listener, make sure both are called when template updates
  - remove first listener, make sure only second listener is called
  - add third/fourth listener, make sure 2nd/3rd/4th are called
  - remove 3rd listener, make sure 2nd/4th are called
  - remove all listeners, make sure none are called


All of this is predicated on the ability to trigger the listener, which is based on the assumption we'll have a function+admin-sdk deployed on the project as an enabler - once that's up, the ability to e2e test this thing should be pretty thorough